### DISCUSSION TITLE
Custom Navigation

### DISCUSSION THREAD

**Discussion by ChuckJonas** [keystonejs/keystone#6131] [Feature requests]
Need more control over navigation and support for sub navigation:

- Control over order
- Custom navigation items
- Group items (Sub-navigation) 

Schema might look something like this

```
{ listItem: "blogPost" },
{
   label: "sales",
   items: {
       { listItem: "Account" },
       { listItem: "Order" },
       { label: "Shopify", linkTo: "https://myAccount.shopify.com" },
   }
},
{ label: "Report Issue", linkTo: "https://myCompany.support.com" },
```

**Comment by raveling** [comment]
@ChuckJonas

I assume you're talking about Admin UI sidebar primarily?

If so, we recently shipped some improvements aligned with your suggestion:

https://keystonejs.com/docs/guides/custom-admin-ui-navigation
https://keystonejs.com/docs/guides/custom-admin-ui-pages

Grouping patterns (in nav sidebar, and grouping fields) is an area we're designing for now. Release timeline unknown at this point in time.

Hope this helps!

**Reply by nickmcmillan** [reply]
Hi @raveling, has any progress been made re: grouping list items in the nav sidebar?
Or could this otherwise be achieved by writing a custom admin UI nav?
Thanks!
